src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -93,8 +108,24 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
     # if the user had not logged in.
     check_kube_connection(configuration)
 
+    required_node_exists = check_linux_amd64_node(configuration)
+    if not required_node_exists:
+        telemetry.set_user_fault()
+        telemetry.set_exception(fault_type=consts.Linux_Amd64_Node_Not_Exists,
+                                summary="Couldn't find any node on the kubernetes cluster with the architecture type 'amd64' and OS 'linux'")
+        raise CLIError("Couldn't connect this Kubernetes cluster to Azure Arc as it doesn't have any nodes with OS 'linux' and architecture 'amd64'. Learn more at {}".format("https://aka.ms/ArcK8sSupportedOSArchitecture"))
+
     # Get kubernetes cluster info
     kubernetes_version = get_server_version(configuration)
+    current_version_minor = kubernetes_version.split('.')[1]
+    try:
+        latest_kubernetes_version = utils.get_latest_kubernetes_version()
+        latest_version_minor = latest_kubernetes_version.split('.')[1]
+        if int(current_version_minor) < int(latest_version_minor) - 2:
+            logger.warning("The Kubernetes cluster you are trying to connect to Azure Arc is of version {}, which is out of Azure Arc's support window. Learn more at {}".format(kubernetes_version, "https://aka.ms/ArcK8sVersionSupportPolicy"))
+    except Exception as e:
```

Comment:
There is a flaw in the check when the major version changes. Like currently when 1.20.0 releases and current version is 1.19.3..

Reply:
Major.Minor.Patch is the semantic versioning svheme right?

I missed one check here. Updating it to, for out of support window criteria.
`if (int(current_version_major) < int(latest_version_major)) or (int(current_version_minor) < int(latest_version_minor) - 2):`